integration-tests/src/test/scala/za/co/absa/spline/harvester/SparkUnimplementedCommandsSpec.scala

```diff
@@ -0,0 +1,285 @@
+
+        withLineageTracking(spark) { lineageCaptor =>
+          val (plan, _) = lineageCaptor.lineageOf {
+            spark.sql(s"TRUNCATE TABLE $tableName") // TruncateTableCommand
+          }
+        }
+      }
+    }
+
+  "Lineage for alter table add columns" should "be caught" in
+    withCustomSparkSession(_
+    .enableHiveSupport()
+    .config("hive.exec.dynamic.partition.mode", "nonstrict")) { spark =>
+
+    withHiveDatabase(spark)(databaseName,
+      (tableName, "(x String, ymd int)", Seq(("Tata", 20190401), ("Tere", 20190403)))) {
+
+      withLineageTracking(spark) { lineageCaptor =>
+        val (plan, _) = lineageCaptor.lineageOf {
+          spark.sql(s"ALTER TABLE $tableName ADD COLUMNS (foo int)") // AlterTableAddColumnsCommand
+        }
+      }
+    }
+  }
+
+  /**
+   * column name/type change not supported in spark 2.3 only comment change is supported
+   */
+  "Lineage for alter table change column" should "be caught" in
+    withCustomSparkSession(_
+    .enableHiveSupport()
+    .config("hive.exec.dynamic.partition.mode", "nonstrict")) { spark =>
+
+    withHiveDatabase(spark)(databaseName,
+      (tableName, "(x String, ymd int)", Seq(("Tata", 20190401), ("Tere", 20190403)))) {
+
+      withLineageTracking(spark) { lineageCaptor =>
+        val (plan, _) = lineageCaptor.lineageOf {
+          // AlterTableChangeColumnCommand
+          spark.sql(s"ALTER TABLE $tableName CHANGE COLUMN x x String COMMENT 'This is a comment'")
+        }
+      }
+    }
+  }
+
+  "Lineage for alter table rename" should "be caught" in
+    withCustomSparkSession(_
+    .enableHiveSupport()
+    .config("hive.exec.dynamic.partition.mode", "nonstrict")) { spark =>
+
+    withHiveDatabase(spark)(databaseName,
+      (tableName, "(x String, ymd int)", Seq(("Tata", 20190401), ("Tere", 20190403)))) {
+
+      withLineageTracking(spark) { lineageCaptor =>
+        val (plan, _) = lineageCaptor.lineageOf {
+          spark.sql(s"ALTER TABLE $tableName RENAME TO new_name") // AlterTableRenameCommand
+        }
+      }
+    }
+  }
+
+  "Lineage for load data" should "be caught" in
+    withCustomSparkSession(_
+      .enableHiveSupport()
+      .config("hive.exec.dynamic.partition.mode", "nonstrict")) { spark =>
+
+      withHiveDatabase(spark)(databaseName,
+        (tableName, "(x String, ymd int)", Seq(("Tata", 20190401), ("Tere", 20190403)))) {
+
+        val filePath = tempDirPath + "/loadData.txt"
+
+        printToFile(filePath) { p =>
+          val separator = "\u0001"
+          p.println("FooBar" + separator + "42")
+          p.println("BleBla" + separator + "66")
+        }
+
+        withLineageTracking(spark) { lineageCaptor =>
+          val (plan, _) = lineageCaptor.lineageOf {
+            spark.sql(s"LOAD DATA LOCAL INPATH '$filePath' INTO TABLE $tableName") // LoadDataCommand
+          }
+        }
+      }
+    }
+
+  private def printToFile(filePath: String)(op: java.io.PrintWriter => Unit) {
```

Comment:
Please take a look at `org.apache.commons.io.FileUtils` class. It should contain a method that you can reuse.